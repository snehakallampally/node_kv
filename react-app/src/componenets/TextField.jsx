import { forwardRef, useEffect, useState } from "react";
const TextField = forwardRef((props, ref) => {
  const [error, setError] = useState(false);

  // useEffect(() => {
  //   if (text.length > 10) setError(true);
  //   else setError(false);
  //   console.log(text)
  //   if(props.changeUserName){
  //     props.changeUserName(text)
  //   }
  // }, [text]);

  const onChange = (e) => {
    if (props.onChange) {
      props.onChange(e.target.value, props.field);
    }
    console.log(e.target.value, props.field);
  };
  return (
    <span className="ip">
      <label htmlFor="uname">{props.label}</label>

      <input
        style={props.error ? { borderColor: "red" } : {}}
        className="ip"
        disabled={props.isDisabled}
        type={props.type}
        name="username"
        id="uname"
        ref={ref}
        field={props.field}
        value={props.value}
        onChange={onChange}
        placeholder={props.placeholder}
      />
    </span>
  );
});
export default TextField;
