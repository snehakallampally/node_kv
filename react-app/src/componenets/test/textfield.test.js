import React from "react";
import {
  getByTestId,
  getByText,
  render,
  fireEvent,
} from "@testing-library/react";
import TextField from "../TextField";

describe("Check if input field is working properly", () => {
  test("Check if input is rendering", () => {
    const { getByTestId } = render(<TextField />);

    const element = getByTestId("inputfield-testid");
    expect(element).toBeTruthy();
  });

  test("Check if placeholder is correct", () => {
    const val = "Email";
    const { getByPlaceholderText } = render(<TextField placeholder={val} />);

    const element = getByPlaceholderText(val);
    console.log(element);
    expect(element).toBeTruthy();
  });

  test("Check if onchange is being called", () => {
    const onChange = jest.fn();

    const val = "Sneha";
    const { getByTestId } = render(<TextField onChange={onChange} />);
    const element = getByTestId("inputfield-testid");

    fireEvent.change(element, {
      target: { value: val },
    });

    expect(onChange).toHaveBeenCalled();
  });

  test("Check if value is being updated properly onchange", () => {
    const onChange = jest.fn();

    const val = "Sneha";
    const { getByTestId } = render(<TextField onChange={onChange} />);
    const element = getByTestId("inputfield-testid");

    fireEvent.change(element, {
      target: { value: val },
    });
    expect(element).toHaveValue(val);
  });

  test("check if snapshots are matching properly", () => {
    const onChange = jest.fn();
    const text = "Click";

    const { asFragment } = render(<TextField placeholder={text} onChange={onChange} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
