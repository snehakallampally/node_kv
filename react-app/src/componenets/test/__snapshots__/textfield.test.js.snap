// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if input field is working properly check if snapshots are matching properly 1`] = `
<DocumentFragment>
  <span
    class="ip"
  >
    <label
      for="uname"
    />
    <input
      class="ip"
      data-testid="inputfield-testid"
      id="uname"
      name="username"
      placeholder="Click"
      value=""
    />
  </span>
</DocumentFragment>
`;
